From 3b02350b9e9b021b401762b606f7fb6205dc5dc1 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Fri, 20 Oct 2017 20:36:54 +0900
Subject: [PATCH 1/2] Extract libarchive contents with a file extension

Some of the commands running on extracted content, like javap, require a
specific file extension to work, so the original extension is better
preserved.
---
 diffoscope/comparators/utils/libarchive.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/diffoscope/comparators/utils/libarchive.py b/diffoscope/comparators/utils/libarchive.py
index db953a5..0f8cb4b 100644
--- a/diffoscope/comparators/utils/libarchive.py
+++ b/diffoscope/comparators/utils/libarchive.py
@@ -222,6 +222,8 @@ class LibarchiveContainer(Archive):
                 # Keep directory sizes small. could be improved but should be
                 # good enough for "ordinary" large archives.
                 dst = os.path.join(tmpdir, str(idx // 4096), str(idx % 4096))
+                root, ext = os.path.splitext(entry.pathname)
+                dst += ext
                 # Maintain a mapping of archive path to the extracted path,
                 # avoiding the need to sanitise filenames.
                 self._members[entry.pathname] = dst
-- 
2.14.1

